Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -256,11 +253,10 @@ struct OrdersUpsertUseCase {
     /// Updates, inserts, or prunes the provided StorageShippingLine's taxes using the provided read-only ShippingLine
     ///
     private func handleShippingLineTaxes(_ readOnlyItem: Networking.ShippingLine, _ storageItem: Storage.ShippingLine, _ storage: StorageType) {
-        let shippingID = readOnlyItem.shippingID
 
         // Upsert the taxes from the read-only orderItem
         for readOnlyTax in readOnlyItem.taxes {
-            if let existingStorageTax = storage.loadShippingLineTax(shippingID: shippingID, taxID: readOnlyTax.taxID) {
+            if let existingStorageTax = storageItem.taxes?.first(where: { $0.taxID == readOnlyTax.taxID }) {
```

Comment:
I see that `loadShippingLineTax` method checks for both `shippingID` and `taxID`. Do we have to add the additional check of `shippingID` here?

Reply:
The `storageItem` is sent from `handleOrderShippingLines`, which is the stored shipping line filtered by `shippingID`. That's why checking for `shippingID` here would be redundant.